ast35/Python/ast.c

```diff
@@ -1447,6 +1469,13 @@ ast_for_arguments(struct compiling *c, const node *n)
                 if (TYPE(ch) == COMMA) {
                     int res = 0;
                     i += 2; /* now follows keyword only arguments */
+
+                    if (TYPE(CHILD(n, i)) == TYPE_COMMENT) {
+                        ast_error(c, CHILD(n, i),
+                                "bare * has associated type comment");
+                        return NULL;
```

Comment:
Oh, ha, I see you do cover this case when the bare `*` has a comma after it.

I believe there's still the case of
```
def f(foo,
      * # type: int
      ):
```